### PR TITLE
Reactive foundation tabs

### PR DESCRIPTION
**Pull request by nathf**
Some of these components need their naming convention evaluated.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -185,6 +185,52 @@
 
     <hr/>
 
+		<uxtabarea>
+			<uxtablinkcontainer>
+				<uxtablink controls="firstTab" active="true">Tab One</uxtablink>
+				<uxtablink controls="secondTab">Tab Two</uxtablink>
+				<uxtablink controls="thirdTab">Tab Three</uxtablink>
```
Comment (pvshum) [inline public/index.html]: Something to discuss, should we just rely in order and maintain **id** inernally instead of setting it explicitly ?

Comment (nathf) [inline public/index.html]: Setting it explicitly provides flexibility in how you structure it. Bringing it internal will require a strict order. I'm not so sure we want to force that on people? Obviously it _should_ be rare that they are out of order.

Comment (pvshum) [inline public/index.html]: Not sure if we need a feature shuffling the order?

Comment (ScottMaclure) [inline public/index.html]: Agree - we can say it's a convention that the tab link order must match the tab pane order. Unless we have a use case for it. Kinda like the TDD approach, we don't add the feature until we need it?

Edit: Also, @pvshum what is "inernally"?

File: src/components/uxtablinkcontainer/uxtablinkcontainer.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Ractive.extend({
+	template: RactiveF.templates.uxtablinkcontainer,
+	oninit: function () {
+		this.on('*.changeTab', function (event) {
+			var components = this.findAllComponents('uxtablink');
+			var guid = this._guid;
+
+			_(components)
```
Comment (ScottMaclure) [inline src/components/uxtablinkcontainer/uxtablinkcontainer.js]: Created #33 to track.

**Comment by nathf** [comment]
The tabs no longer require matching defined attributes to function.

Have updated the naming convention of some components
`uxtablinkcontainer` -> `uxtablinks`
`uxtabcontent` -> `uxtabpanes`

I think this will make a bit more sense when we tackle encapsulation.

Also added the mixin for `findAllChildComponents` to app.js. Issue: #33

File: src/components/uxtabarea/uxtabarea.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+		var tabLinks = this.findComponent('uxtablinks');
+		var tabPanes = this.findComponent('uxtabpanes');
+
+		var tabLink = tabLinks.findAllChildComponents('uxtablink');
+		var tabPane = tabPanes.findAllChildComponents('uxtabpane');
+
+		_.each(tabLink, function (link, i) {
+			var childPane;
+			if (childPane = tabPane[i]) {
+				childPane.set('uidLink', link._guid);
```
Comment (ScottMaclure) [inline src/components/uxtabarea/uxtabarea.js]: Q) Where is `uidLink` referenced?

Comment (nathf) [inline src/components/uxtabarea/uxtabarea.js]: It doesn't. Was just making sure you were reading the codes ;)

File: src/components/uxtabarea/uxtabarea.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+		var tabLinks = this.findComponent('uxtablinks');
+		var tabPanes = this.findComponent('uxtabpanes');
+
+		var tabLink = tabLinks.findAllChildComponents('uxtablink');
+		var tabPane = tabPanes.findAllChildComponents('uxtabpane');
+
+		_.each(tabLink, function (link, i) {
+			var childPane;
+			if (childPane = tabPane[i]) {
```
Comment (ScottMaclure) [inline src/components/uxtabarea/uxtabarea.js]: How did this get pass "boss" option??

http://jshint.com/docs/options/#boss

https://github.com/themacclesoft/ractive-foundation/blob/master/.jshintrc#L38

Created issue #37 to track adding jshint to gulp build.

**Pull request merged**